src/roswire/definitions/package.py

```diff
@@ -114,6 +114,43 @@ def paths(shell: dockerblade.Shell,
             paths.extend(package_dirs)
         return paths
 
+    @staticmethod
+    def _paths_ros2(shell: dockerblade.Shell,
+                    files: dockerblade.FileSystem
+                    ) -> List[str]:
+        """Returns paths of packages"""
+        paths: List[str] = []
+        command = "ros2 pkg list"
+        try:
+            package_str = shell.check_output(command, text=True)
+        except dockerblade.exceptions.CalledProcessError:
+            logger.debug(f'unable to find packages using {command}')
+            raise
+        all_packages = package_str.split('\r\n')
+        package_dirs: List[str] = []
+        for p in all_packages:
+            command = 'ros2 pkg prefix ' + p
+            try:
+                package_path = shell.check_output(command, text=True)
+            except dockerblade.exceptions.CalledProcessError:
+                logger.debug(f'unable to find package {p}')
+                raise
+            package_dirs.append(package_path)
```

Comment:
This is a subtle bug. `package_dirs` isn't clearly between loop iterations, and so each time `.extend` is called on the line below, every existing path is reinserted into `paths`. The fix here is to just drop `package_dirs` entirely and append to `paths`.